ext/rotoscope/rotoscope.c

```diff
@@ -78,10 +80,11 @@ static rs_class_desc_t class2str(VALUE klass) {
   } else {
     if (FL_TEST(klass, FL_SINGLETON)) {
       real_class.name = singleton2str(klass);
-      if (is_class_singleton(klass)) {
+      if (is_class_singleton(klass) && rb_check_id(&real_class.name)) {
```

Comment:
`rb_check_id` returns 0 when it's an anonymous instance, so we can leverage that to determine if it's an instance or a class at this point.